samples/SocialSample/SignInExtensions.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authentication.Google;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+
+namespace SocialSample
+{
+    public static class SignInExtensions {
+        // Adds google auth and a default cookie auth for sign in purposes
+        public static AuthenticationBuilder UseGoogleSignIn(this AuthenticationBuilder builder, Action<GoogleOptions> configureOptions)
+        {
+            builder.AddGoogle(GoogleDefaults.AuthenticationScheme, o =>
+            {
+                configureOptions?.Invoke(o);
+                // Override instead of default since this method is opinionated on the cookie scheme name.
+                o.SignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+            });
+
+            builder.ConfigureCookie(GoogleDefaults.AuthenticationScheme);
+            return builder;
+        }
+
+        // Add the defalut cookie if needed, set it to default and configure the cookie to forward challenge
+        private static void ConfigureCookie(this AuthenticationBuilder builder, string forwardChallenge)
```

Comment:
Aren't these also supposed to set cookies as the overall default scheme? Or how else does it work with `[Authroize]`

Reply:
It does on line 31:

`o.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;`

Reply:
Ah, I mistook that for the new virtual DefaultScheme.